From 6070f54d462367268c480a6d10dfddd2af67f3d5 Mon Sep 17 00:00:00 2001
From: Jason Rogena <j.rogena@cgiar.org>
Date: Mon, 31 Mar 2014 15:23:52 +0300
Subject: [PATCH 135/213] Version 0.8 Revision 10: all activities now save the
 values of their edit texts when activity is paused and restores the data when
 the activity resumes

---
 MistroFarmer/src/main/AndroidManifest.xml          |  2 +-
 .../cgiar/ilri/mistro/farmer/AddEventActivity.java | 24 ++++++++++++++++++
 .../mistro/farmer/AddMilkProductionActivity.java   | 14 +++++++++++
 .../mistro/farmer/CowRegistrationActivity.java     | 29 ++++++++++++++++++++++
 .../mistro/farmer/FarmerRegistrationActivity.java  | 10 ++++++++
 .../cgiar/ilri/mistro/farmer/LandingActivity.java  |  2 +-
 .../ilri/mistro/farmer/backend/DataHandler.java    | 24 ++++++++++++++++++
 7 files changed, 103 insertions(+), 2 deletions(-)

diff --git a/MistroFarmer/src/main/AndroidManifest.xml b/MistroFarmer/src/main/AndroidManifest.xml
index 05979c9..8de8528 100644
--- a/MistroFarmer/src/main/AndroidManifest.xml
+++ b/MistroFarmer/src/main/AndroidManifest.xml
@@ -2,7 +2,7 @@
 <manifest xmlns:android="http://schemas.android.com/apk/res/android"
     package="org.cgiar.ilri.mistro.farmer"
     android:versionCode="2"
-    android:versionName="0.8.9" >
+    android:versionName="0.8.10" >
 
     <uses-sdk
         android:minSdkVersion="7"
diff --git a/MistroFarmer/src/main/java/org/cgiar/ilri/mistro/farmer/AddEventActivity.java b/MistroFarmer/src/main/java/org/cgiar/ilri/mistro/farmer/AddEventActivity.java
index 921f8a9..8c30132 100644
--- a/MistroFarmer/src/main/java/org/cgiar/ilri/mistro/farmer/AddEventActivity.java
+++ b/MistroFarmer/src/main/java/org/cgiar/ilri/mistro/farmer/AddEventActivity.java
@@ -159,6 +159,30 @@ public class AddEventActivity extends SherlockActivity implements View.OnClickLi
         }
     }
 
+    @Override
+    protected void onPause() {
+        super.onPause();
+        DataHandler.setSharedPreference(this, DataHandler.SP_KEY_AEA_DATE, dateET.getText().toString());
+        DataHandler.setSharedPreference(this, DataHandler.SP_KEY_AEA_REMARKS, remarksET.getText().toString());
+        DataHandler.setSharedPreference(this, DataHandler.SP_KEY_AEA_STRAW_NUMBER, strawNumberET.getText().toString());
+        DataHandler.setSharedPreference(this, DataHandler.SP_KEY_AEA_VET_USED, vetUsedET.getText().toString());
+        DataHandler.setSharedPreference(this, DataHandler.SP_KEY_AEA_BULL_NAME, bullNameACTV.getText().toString());
+        DataHandler.setSharedPreference(this, DataHandler.SP_KEY_AEA_BULL_OWNER, specBullOwnerET.getText().toString());
+        DataHandler.setSharedPreference(this, DataHandler.SP_KEY_AEA_NO_SERVICING_DAYS, noOfServicingDaysET.getText().toString());
+    }
+
+    @Override
+    protected void onResume() {
+        super.onResume();
+        dateET.setText(DataHandler.getSharedPreference(this, DataHandler.SP_KEY_AEA_DATE, ""));
+        remarksET.setText(DataHandler.getSharedPreference(this, DataHandler.SP_KEY_AEA_REMARKS, ""));
+        strawNumberET.setText(DataHandler.getSharedPreference(this, DataHandler.SP_KEY_AEA_STRAW_NUMBER, ""));
+        vetUsedET.setText(DataHandler.getSharedPreference(this, DataHandler.SP_KEY_AEA_VET_USED, ""));
+        bullNameACTV.setText(DataHandler.getSharedPreference(this, DataHandler.SP_KEY_AEA_BULL_NAME, ""));
+        specBullOwnerET.setText(DataHandler.getSharedPreference(this, DataHandler.SP_KEY_AEA_BULL_OWNER, ""));
+        noOfServicingDaysET.setText(DataHandler.getSharedPreference(this, DataHandler.SP_KEY_AEA_NO_SERVICING_DAYS, ""));
+    }
+
     public boolean onCreateOptionsMenu(Menu menu) {
         MenuInflater inflater = getSupportMenuInflater();
         inflater.inflate(R.menu.add_event, menu);
diff --git a/MistroFarmer/src/main/java/org/cgiar/ilri/mistro/farmer/AddMilkProductionActivity.java b/MistroFarmer/src/main/java/org/cgiar/ilri/mistro/farmer/AddMilkProductionActivity.java
index 6d69821..41c87d7 100644
--- a/MistroFarmer/src/main/java/org/cgiar/ilri/mistro/farmer/AddMilkProductionActivity.java
+++ b/MistroFarmer/src/main/java/org/cgiar/ilri/mistro/farmer/AddMilkProductionActivity.java
@@ -93,6 +93,20 @@ public class AddMilkProductionActivity extends SherlockActivity implements View.
         fetchCowIdentifiers();
     }
 
+    @Override
+    protected void onPause() {
+        super.onPause();
+        DataHandler.setSharedPreference(this, DataHandler.SP_KEY_AMPA_DATE, dateET.getText().toString());
+        DataHandler.setSharedPreference(this, DataHandler.SP_KEY_AMPA_QUANTITY, quantityET.getText().toString());
+    }
+
+    @Override
+    protected void onResume() {
+        super.onResume();
+        dateET.setText(DataHandler.getSharedPreference(this, DataHandler.SP_KEY_AMPA_DATE, ""));
+        quantityET.setText(DataHandler.getSharedPreference(this, DataHandler.SP_KEY_AMPA_QUANTITY, ""));
+    }
+
     public boolean onCreateOptionsMenu(com.actionbarsherlock.view.Menu menu) {
         MenuInflater inflater = getSupportMenuInflater();
         inflater.inflate(R.menu.milk_production, menu);
diff --git a/MistroFarmer/src/main/java/org/cgiar/ilri/mistro/farmer/CowRegistrationActivity.java b/MistroFarmer/src/main/java/org/cgiar/ilri/mistro/farmer/CowRegistrationActivity.java
index b4e2dff..978e679 100644
--- a/MistroFarmer/src/main/java/org/cgiar/ilri/mistro/farmer/CowRegistrationActivity.java
+++ b/MistroFarmer/src/main/java/org/cgiar/ilri/mistro/farmer/CowRegistrationActivity.java
@@ -208,9 +208,38 @@ public class CowRegistrationActivity extends SherlockActivity implements View.On
     }
 
     @Override
+    protected void onPause() {
+        super.onPause();
+
+        //Incase the activity is hidden partially/fully save the data in edittexts
+        DataHandler.setSharedPreference(this, DataHandler.SP_KEY_CRA_NAME, nameET.getText().toString());
+        DataHandler.setSharedPreference(this, DataHandler.SP_KEY_CRA_EAR_TAG_NUMBER, earTagNumberET.getText().toString());
+        DataHandler.setSharedPreference(this, DataHandler.SP_KEY_CRA_AGE, ageET.getText().toString());
+        DataHandler.setSharedPreference(this, DataHandler.SP_KEY_CRA_DATE_OF_BIRTH, dateOfBirthET.getText().toString());
+        DataHandler.setSharedPreference(this, DataHandler.SP_KEY_CRA_BREED, breedET.getText().toString());
+        DataHandler.setSharedPreference(this, DataHandler.SP_KEY_CRA_DEFORMITY, deformityET.getText().toString());
+        DataHandler.setSharedPreference(this, DataHandler.SP_KEY_CRA_STRAW_NUMBER, strawNumberET.getText().toString());
+        DataHandler.setSharedPreference(this, DataHandler.SP_KEY_CRA_DAM, damACTV.getText().toString());
+        DataHandler.setSharedPreference(this, DataHandler.SP_KEY_CRA_EMBRYO_NUMBER, embryoNumberET.getText().toString());
+        DataHandler.setSharedPreference(this, DataHandler.SP_KEY_CRA_COUNTRY_OF_ORIGIN, countryOfOriginACTV.getText().toString());
+    }
+
+    @Override
     protected void onResume() {
         super.onResume();
 
+        //incase the activity was hidden partially for a moment, restore what the user had already entered
+        nameET.setText(DataHandler.getSharedPreference(this, DataHandler.SP_KEY_CRA_NAME, ""));
+        earTagNumberET.setText(DataHandler.getSharedPreference(this, DataHandler.SP_KEY_CRA_EAR_TAG_NUMBER, ""));
+        ageET.setText(DataHandler.getSharedPreference(this, DataHandler.SP_KEY_CRA_AGE, ""));
+        dateOfBirthET.setText(DataHandler.getSharedPreference(this, DataHandler.SP_KEY_CRA_DATE_OF_BIRTH, ""));
+        breedET.setText(DataHandler.getSharedPreference(this, DataHandler.SP_KEY_CRA_BREED, ""));
+        deformityET.setText(DataHandler.getSharedPreference(this, DataHandler.SP_KEY_CRA_DEFORMITY, ""));
+        strawNumberET.setText(DataHandler.getSharedPreference(this, DataHandler.SP_KEY_CRA_STRAW_NUMBER, ""));
+        damACTV.setText(DataHandler.getSharedPreference(this, DataHandler.SP_KEY_CRA_DAM, ""));
+        embryoNumberET.setText(DataHandler.getSharedPreference(this, DataHandler.SP_KEY_CRA_EMBRYO_NUMBER, ""));
+        countryOfOriginACTV.setText(DataHandler.getSharedPreference(this, DataHandler.SP_KEY_CRA_COUNTRY_OF_ORIGIN, ""));
+
         Bundle bundle=this.getIntent().getExtras();
         if(bundle != null) {
             farmer=bundle.getParcelable(Farmer.PARCELABLE_KEY);
diff --git a/MistroFarmer/src/main/java/org/cgiar/ilri/mistro/farmer/FarmerRegistrationActivity.java b/MistroFarmer/src/main/java/org/cgiar/ilri/mistro/farmer/FarmerRegistrationActivity.java
index 371a4d0..622d35f 100644
--- a/MistroFarmer/src/main/java/org/cgiar/ilri/mistro/farmer/FarmerRegistrationActivity.java
+++ b/MistroFarmer/src/main/java/org/cgiar/ilri/mistro/farmer/FarmerRegistrationActivity.java
@@ -109,6 +109,11 @@ public class FarmerRegistrationActivity extends SherlockActivity implements View
     protected void onResume()
     {
         super.onResume();
+
+        fullNameET.setText(DataHandler.getSharedPreference(this, DataHandler.SP_KEY_FRA_FULL_NAME, ""));
+        extensionPersonnelET.setText(DataHandler.getSharedPreference(this, DataHandler.SP_KEY_FRA_EXTENSION_PERSONNEL, ""));
+        mobileNumberET.setText(DataHandler.getSharedPreference(this, DataHandler.SP_KEY_FRA_MOBILE_NUMBER, ""));
+
         Bundle bundle=this.getIntent().getExtras();
         if(bundle!=null)
         {
@@ -141,6 +146,11 @@ public class FarmerRegistrationActivity extends SherlockActivity implements View
     protected void onPause()
     {
         super.onPause();
+
+        DataHandler.setSharedPreference(this, DataHandler.SP_KEY_FRA_FULL_NAME, fullNameET.getText().toString());
+        DataHandler.setSharedPreference(this, DataHandler.SP_KEY_FRA_EXTENSION_PERSONNEL, extensionPersonnelET.getText().toString());
+        DataHandler.setSharedPreference(this, DataHandler.SP_KEY_FRA_MOBILE_NUMBER, mobileNumberET.getText().toString());
+
         if(locationManager!=null)
         {
             locationManager.removeUpdates(this);
diff --git a/MistroFarmer/src/main/java/org/cgiar/ilri/mistro/farmer/LandingActivity.java b/MistroFarmer/src/main/java/org/cgiar/ilri/mistro/farmer/LandingActivity.java
index b53083b..c5ddb22 100644
--- a/MistroFarmer/src/main/java/org/cgiar/ilri/mistro/farmer/LandingActivity.java
+++ b/MistroFarmer/src/main/java/org/cgiar/ilri/mistro/farmer/LandingActivity.java
@@ -77,7 +77,7 @@ public class LandingActivity extends SherlockActivity implements View.OnClickLis
 
         //init text according to locale
         initTextInViews();
-        Toast.makeText(this, "Version 0.8 Revision 9", Toast.LENGTH_LONG).show();
+        Toast.makeText(this, "Version 0.8 Revision 10", Toast.LENGTH_LONG).show();
     }
 
     public boolean onCreateOptionsMenu(Menu menu) {
diff --git a/MistroFarmer/src/main/java/org/cgiar/ilri/mistro/farmer/backend/DataHandler.java b/MistroFarmer/src/main/java/org/cgiar/ilri/mistro/farmer/backend/DataHandler.java
index fa8749f..ec8075f 100644
--- a/MistroFarmer/src/main/java/org/cgiar/ilri/mistro/farmer/backend/DataHandler.java
+++ b/MistroFarmer/src/main/java/org/cgiar/ilri/mistro/farmer/backend/DataHandler.java
@@ -76,6 +76,30 @@ public class DataHandler
     public static final String SP_KEY_USE_SMS_TO_SEND_DATA = "useSMSToSendData";
     public static final String SP_KEY_SMS_RESPONSE = "smsResponse";
     public static final String SP_KEY_SMS_CACHE = "smsCache";
+
+    public static final String SP_KEY_AEA_DATE = "addEventActivityDate";
+    public static final String SP_KEY_AEA_REMARKS = "addEventActivityRemarks";
+    public static final String SP_KEY_AEA_STRAW_NUMBER = "addEventActivityStrawNumber";
+    public static final String SP_KEY_AEA_VET_USED = "addEventActivityVetUsed";
+    public static final String SP_KEY_AEA_BULL_NAME = "addEventActivityBullName";
+    public static final String SP_KEY_AEA_BULL_OWNER = "addEventActivityBullOwner";
+    public static final String SP_KEY_AEA_NO_SERVICING_DAYS = "addEventActivityNoServicingDays";
+    public static final String SP_KEY_AMPA_DATE = "addMilkProductionActivityDate";
+    public static final String SP_KEY_AMPA_QUANTITY = "addMilkProductionActivityQuantity";
+    public static final String SP_KEY_CRA_NAME = "cowRegistrationActivityName";
+    public static final String SP_KEY_CRA_EAR_TAG_NUMBER = "cowRegistrationActivityEarTagNumber";
+    public static final String SP_KEY_CRA_AGE = "cowRegistrationActivityAge";
+    public static final String SP_KEY_CRA_DATE_OF_BIRTH = "cowRegistrationActivityDateOfBirth";
+    public static final String SP_KEY_CRA_BREED = "cowRegistrationActivityBreed";
+    public static final String SP_KEY_CRA_DEFORMITY = "cowRegistrationActivityDeformity";
+    public static final String SP_KEY_CRA_STRAW_NUMBER = "cowRegistrationActivityStrawNumber";
+    public static final String SP_KEY_CRA_DAM = "cowRegistrationActivityDam";
+    public static final String SP_KEY_CRA_EMBRYO_NUMBER = "cowRegistrationActivityEmbryoNumber";
+    public static final String SP_KEY_CRA_COUNTRY_OF_ORIGIN = "cowRegistrationActivityCountryOfOrigin";
+    public static final String SP_KEY_FRA_FULL_NAME = "farmerRegistrationActivityFullName";
+    public static final String SP_KEY_FRA_EXTENSION_PERSONNEL = "farmerRegistrationActivityExtensionPersonnel";
+    public static final String SP_KEY_FRA_MOBILE_NUMBER = "farmerRegistrationActivityMobileNumber";
+
     public static final String CAN_SEND_USING_SMS = "canSend";
     public static final String CANNOT_SEND_USING_SMS = "cannotSend";
     public static final String SMS_DELIMITER = "#*#*";
-- 
2.7.1

